src/openst.js

```diff
@@ -68,9 +70,57 @@ class OpenST {
       createAndAddModules: createAndAddModules.address,
       tokenRules: tokenRulesAddress,
     };
+
     Object.assign(this.chainConfig.openst, setupData);
+
+    this.chainConfig.utilityBrandedTokens[index].openst = setupData;
+
     logger.info('Completed Setup of OpenST');
   }
+
+  getUtilityBrandedTokenConfig(auxiliaryEIP20Token) {
+    for (let i = 0; i < this.chainConfig.utilityBrandedTokens.length; i += 1) {
+      const ut = this.chainConfig.utilityBrandedTokens[i];
+      if (ut.address === auxiliaryEIP20Token) {
+        return { ubt: ut, index: i };
+      }
+    }
+    return { ubt: undefined, index: -1 };
+  }
+
+  async deployPricerRule(
+    auxiliaryEIP20Token,
+    baseCurrencyCode,
+    conversionRate,
+    conversionRateDecimals,
+    requiredPriceOracleDecimals,
+  ) {
+    logger.info('Deployment of PricerRule');
+    const ubtConfig = this.getUtilityBrandedTokenConfig(auxiliaryEIP20Token);
+
+    const pricerRule = await PricerRule.deploy(
+      this.auxiliary.web3,
+      ubtConfig.ubt.organizationAddress,
+      ubtConfig.ubt.address,
+      ubtConfig.ubt.openst.tokenRules,
+      baseCurrencyCode,
+      conversionRate,
+      conversionRateDecimals,
+      requiredPriceOracleDecimals,
+      this.auxiliary.txOptions,
+    );
+
+    const pricerOracleData = {
+      [baseCurrencyCode]: {
+        address: pricerRule.address,
+      },
+    };
+
+    this.chainConfig.utilityBrandedTokens[ubtConfig.index].pricerOracles = pricerOracleData;
```

Comment:
As per current logic since there can be only one pricerRule contract per economy we can update this address in **.openst** object. What do you think?